receiver/httpcheckreceiver/metadata.yaml

```diff
@@ -0,0 +1,32 @@
+name: httpcheckreceiver
+
+resource_attributes:
+
+attributes:
+  http.url:
+    description: Full HTTP request URL.
+    type: string
+  http.status_code:
+    description: HTTP response status code
+    type: int
+  http.method:
+    description: HTTP request method
+    type: string
+
+metrics:
+  httpcheck.success:
+    description: The response of the endpoint resulted in status 200.
+    enabled: true
+    gauge:
```

Comment:
Right, the way I was thinking about is that using a gauge, users would setup alerting based on individual checks failing rather than thinking about it in terms of aggregates. I can see how modeling this using a sum + delta aggregation would work as well, although this would mean that users couldn't infer meaning from the value being reported without looking at the attributes as well.

Reply:
I see what you're saying about it being difficult to alert on. What if the "status_code: 200" data point is always reported, either as 1 or 0? This would have no impact on aggregations, but also would allow for easy alerting on a specific check.

Reply:
I guess then the question becomes why is `status_code: 200` special? Naming this `success` while combining it with a configuration option to define what success means to users i think would address most use cases.

Using a non-monotonic counter with delta temporality here would still allow users to aggregate data on the `code` to determine what kinds of issues are occurring across multiple endpoints. Imagining the following reports:

```
{ "time": "hour 0", "url": "/foo1", "method": "GET", "code": 200, "value": 1 }
{ "time": "hour 0", "url": "/foo2", "method": "GET", "code": 500, "value": 0 } // outage
{ "time": "hour 1", "url": "/foo1", "method": "GET", "code": 500, "value": 0 } // outage
{ "time": "hour 1", "url": "/foo2", "method": "GET", "code": 500, "value": 0 } // outage
{ "time": "hour 2", "url": "/foo1", "method": "GET", "code": 200, "value": 1 } // recovered
{ "time": "hour 2", "url": "/foo2", "method": "GET", "code": 200, "value": 1 } // recovered
```

Reply:
> I guess then the question becomes why is `status_code: 200` special?

I think @jmacd's [suggestion](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14191#discussion_r987393250) resolves this, by treating none of the values as special and reporting 0's for all but one code. This is a bit verbose but is a well established format called [one-hot encoding](https://en.wikipedia.org/wiki/One-hot#Machine_learning_and_statistics).